cylc/sphinx_ext/cylc_lang/autodocumenters.py

```diff
@@ -346,3 +367,314 @@ def run(self):
         )
 
         return [node]
+
+
+class CylcMetadataDirective(SphinxDirective):
+    """Represent a Cylc Config.
+    """
+    optional_arguments = 3
+
+    def run(self):
+        # Parse input options:
+        for key, value in zip(
+            [i.strip(':') for i in self.arguments[::2]],
+            list(self.arguments[1::2])
+        ):
+            self.options.update({key: value})
+
+        # Get global or workflow metadara
+        if 'source' in self.options:
+            config = self.load_workflow_cfg(self.options['source'])
+            metadata = self.get_workflow_metadata(
+                config, self.options['source'])
+            ret = self.workflow_to_node(metadata, config)
+        else:
+            src = self.options['global']
+            config = self.load_global_cfg(self.options['global'])
+            metadata = self.get_global_metadata(config)
+            ret = self.global_to_node(metadata, src)
+
+        node = addnodes.desc_content()
+        self.state.nested_parse(
+            StringList(ret),
+            self.content_offset,
+            node
+        )
+        return [node]
+
+    @staticmethod
+    def load_global_cfg(conf_path=None):
+        """Get Global Configuration metadata:
+
+        Args:
+            Path: Global conf path.
+
+        """
+        # Load Global Config:
+        if conf_path:
+            env = os.environ
+            sub = run(
+                ['cylc', 'config', '--json'],
+                capture_output=True,
+                env=env.update({'CYLC_SITE_CONF_PATH': conf_path})
+            )
+        else:
+            sub = run(['cylc', 'config', '--json'], capture_output=True)
+
+        CylcMetadataDirective.check_subproc_output(sub)
+
+        return json.loads(sub.stdout)
+
+    @staticmethod
+    def get_global_metadata(config):
+        """
+        Additional Processing:
+        * Get lists of hosts/platforms and job runner from the config.
+        * If no title is provided, use the platform/group regex as the title.
+        * If title != regex then insert text saying which regex
+          needs matching to select this platform.
+
+        Returns:
+            A dictionary in the form:
+            'platforms': {'platform regex': {..metadata..}},
+            'platform groups': {'platform regex': {..metadata..}}
+        """
+        metadata = {}
+        for section, select_from in zip(
+            ['platforms', 'platform groups'],
+            ['hosts', 'platforms']
+        ):
```

Comment:
Looks like we have to walk over these four sections again later anyway.

Might be easier to just walk over the config and write the docs at the same time rather than collecting all the metadata into a dict and then generating the config from the dict. Note, we will want to document other things in the future, env vars, scripts, platforms, etc.